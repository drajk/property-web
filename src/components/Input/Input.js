import React from 'react';
import PropTypes from 'prop-types';
import styles from './Input.module.css';

const Input = ({ onChange, placeholder, value, ...otherProps }) => {
    const inputValue = (value || '').replace(/^\s+/, '');
    const isEmpty = inputValue === '';

    const placeholderElement = placeholder && isEmpty && (
        <span className={styles.placeholder}>{placeholder}</span>
    );

    const handleChange = (event) => {
        event.preventDefault();
        onChange(event.target.value);
    };

    return (
        <div className={styles.wrapper}>
            {placeholderElement}
            <input
                type="text"
                value={inputValue}
                onChange={handleChange}
                className={styles.input}
                {...otherProps}
            />
        </div>
    );
};

Input.propTypes = {
    onChange: PropTypes.func,
    placeholder: PropTypes.string,
    value: PropTypes.string,
};

Input.defaultProps = {
    onChange: () => {},
    placeholder: '',
    value: '',
};

export default Input;
