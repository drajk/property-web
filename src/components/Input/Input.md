# Input.js

Shows a styled input box with placeholder.

## Usage

```html
<Input placeholder="Search" />
<Input placeholder="Search" onChange={()=>{}}/>
<Input placeholder="Search" value="Initial Value" />
```
