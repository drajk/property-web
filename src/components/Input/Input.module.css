.wrapper {
    position: relative;
    overflow: hidden;
    margin-top: 20px;
}

.placeholder {
    position: absolute;
    text-transform: none;
    line-height: 1;
    top: 22px;
    left: 15px;
    pointer-events: none;
    color: var(--grey);
    display: block;
}

.input {
    font-size: var(--font-size-normal);
    padding-top: 0;
    color: var(--color);
    height: 60px;
    border: 1px solid #ccc;
    padding: 0 15px;
    border-radius: 4px;
    width: 300px;
}