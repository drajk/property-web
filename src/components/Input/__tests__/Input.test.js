import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';

describe('Input', () => {
    it('renders with placeholder', () => {
        const tree = renderer.create(<Input placeholder="Search" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with on click handler', () => {
        const tree = renderer
            .create(<Input placeholder="Search" onChange={() => {}} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with initial value', () => {
        const tree = renderer
            .create(<Input placeholder="Search" value="Initial Value" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
