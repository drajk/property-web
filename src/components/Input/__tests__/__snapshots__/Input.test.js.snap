// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders with initial value 1`] = `
<div
  className="wrapper"
>
  <input
    className="input"
    onChange={[Function]}
    type="text"
    value="Initial Value"
  />
</div>
`;

exports[`Input renders with on click handler 1`] = `
<div
  className="wrapper"
>
  <span
    className="placeholder"
  >
    Search
  </span>
  <input
    className="input"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Input renders with placeholder 1`] = `
<div
  className="wrapper"
>
  <span
    className="placeholder"
  >
    Search
  </span>
  <input
    className="input"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
